spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/data/RandomData.java

```diff
@@ -332,6 +337,26 @@ public Object map(Types.MapType map, Supplier<Object> keyResult, Supplier<Object
       return result;
     }
 
+    public VariantVal variant(Types.VariantType variant) {
+      // TIME, TIMESTAMPTZ_NANOS and TIMESTAMPNTZ_NANOS are not supported in Spark
```

Comment:
Aren't these valid in Variant? I don't think that Spark is actually processing these values. In addition, it should not matter to Spark if they are embedded in a variant unless it tries to interpret them. For these tests, this shouldn't be needed.

I tried removing the exceptions and the `SparkParquetReader` tests passed.